Classes/RedirectService.php

```diff
@@ -106,8 +98,24 @@ protected function buildResponse(ServerRequestInterface $httpRequest, RedirectIn
         if ($statusCode >= 300 && $statusCode <= 399) {
             $location = $redirect->getTargetUriPath();
 
+            // Relative redirects will be turned into absolute redirects
             if (parse_url($location, PHP_URL_SCHEME) === null) {
-                $location = (string)$httpRequest->getUri()->withQuery('')->withFragment('')->withPath($location);
+                $locationParts = parse_url($location);
```

Comment:
Seems the ordering is wrong here. `parse_url($location, PHP_URL_SCHEME)` should be null and then its used as locationParts.
In general there are too much `$location` variables. Maybe you could name them more explicit?

Reply:
The first parse url checks whether it's an absolute url. The second the splits it.

Reply:
Ah, ok

Reply:
Still to much $location for my taste ;)

Reply:
Changed ;)